### DISCUSSION TITLE
Mozilla Thunderbird

### DISCUSSION THREAD

**Discussion by Tuharsky** [ltsp/ltsp#462] [Q&A] [answered]
Hi.
Please, is there a way to keep Mozilla Thunderbird working with LTSP? With thin client, that was no problem. With the fat client (such as LTSP2x) the program runs, but the inbox is impossible to open. Probably it has problem with the mapped home folder.

**Comment by alkisg** [comment]
I just tested once more. Thunderbird runs fine on LTSP 2x fat clients.
What problem do you have?
What errors do you see if you run it from a console?

Are you using some weird LTSP setup with the LTSP image on a local disk, or anything else that we should know?

**Reply by Tuharsky** [reply]
Thank You, Alkis.
I think I found the culprit - it is the symbolic link used in the path of the home folder. The fat client's Thunderbird seemingly cannot cope with it. There are no errors on console.
I do have in lts.conf the declaration
SSH_FOLLOW_SYMLINKS = True

**Comment by alkisg** [comment]
LTSP5 had lts.conf which supported SSH_FOLLOW_SYMLINKS.

LTSP20+ has [ltsp.conf](https://ltsp.org/man/ltsp.conf/) where there's no SSH_FOLLOW_SYMLINKS option.

I don't know which "symbolic link used in the path of the home folder" you are referring to; thunderbird saves its data in /home/username/.thunderbird without using any symlinks.

**Reply by Tuharsky** [reply]
Well, this is the "weird" thing in our setup, the symlinked home directories.
As of LTSP5, the fat client's Thunderbird did not work either, but since thin client option was still available, it did not bother me.

**Comment by alkisg** [comment]
You didn't mention *where* they're symlinked...
If you need help, you need to describe your setup in more detail.
If you found a workaround, then there's no need to explain it.

**Reply by Tuharsky** [reply]
Understood. The symlink is at the level of user's home directory. It is the USER part of /home/USER

The interesting thing is, that the Thunderbird profile is mostly functional - the program opens, has all accounts displayed, account settings are OK. Just the mail folders are locked and don't open on click.

In the Thunderbird error console, I get
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no]

**Comment by alkisg** [comment, marked as answer]
I'm going to guess that on the LTSP server, you have a symlink
from /home/username
to /somewhere-else/username

This then shouldn't cause any issues at all since that symlink doesn't exist in LTSP clients, where /home/username it an SSHFS mount over an overlayfs directory, it's not a symlink at all.

BUT if the user has logged into the server at some point, then thunderbird could have stored the real path, /somewhere-else/username/.thunderbird, which of course isn't there in LTSP clients.

I.e. in that case the problem isn't related to LTSP or SSHFS at all.

**Reply by Tuharsky** [reply]
Very likely possibility. I presumed that it is some kind of LTSP redirection problem, but I will definitely better check the Thunderbird internals. Thank You.

**Reply by Tuharsky** [reply]
Alkis, You diagnosed the problem well.

I went to Thunderbird Settings - Editor, then searched for the /home/ string. Found bunch of mail.root.* and mail.server.* preferences, edited the paths, and the Thunderbird works now.

Thank You for advice, hopefully it might help others too.
